ddtrace/internal/ci_visibility/recorder.py

```diff
@@ -72,6 +77,41 @@ def __init__(self, tracer=None, config=None, service=None):
         except Exception:
             log.warning("Failed to load CODEOWNERS", exc_info=True)
 
+    def _configure_writer(self):
+        writer = None
+        if ddconfig._ci_visibility_agentless_enabled:
+            headers = {"dd-api-key": os.environ.get("DD_API_KEY")}
+            if headers["dd-api-key"]:
+                writer = CIVisibilityWriter(
+                    intake_url="%s.%s" % (AGENTLESS_BASE_URL, os.environ.get("DD_SITE", AGENTLESS_DEFAULT_SITE)),
+                    headers=headers,
+                )
+            else:
+                raise EnvironmentError(
+                    "DD_CIVISIBILITY_AGENTLESS_ENABLED is set, but DD_API_KEY is not set, so ddtrace "
+                    "cannot be initialized."
+                )
+        elif self._agent_evp_proxy_is_available():
+            writer = CIVisibilityWriter(
+                intake_url=agent.get_trace_url(),
+                headers={EVP_SUBDOMAIN_HEADER_NAME: EVP_SUBDOMAIN_HEADER_VALUE},
+                use_evp=True,
+            )
+        if writer is not None:
+            self.tracer.configure(writer=writer)
+
+    def _agent_evp_proxy_is_available(self):
+        try:
+            info = agent.info()
+        except Exception:
+            info = None
+
+        if info:
+            endpoints = info.get("endpoints", [])
+            if endpoints and (EVP_PROXY_AGENT_BASE_PATH in endpoints or ("/" + EVP_PROXY_AGENT_BASE_PATH) in endpoints):
```

Comment:
nitpick: would this be more robust? I see that some endpoints in the `/info` response include a final `/` and some don't. 
```suggestion
            if endpoints and any(EVP_PROXY_AGENT_BASE_PATH in endpoint for endpoint in endpoints):
```
for this `EVP_PROXY_AGENT_BASE_PATH` would have to remove the trailing slash `/`